### ISSUE TITLE
audio works, but video needs rejoining to work

### ISSUE THREAD

**Issue by varundabke** [edumeet/edumeet#633]
Hello,
   Many times when a Mobile user joins, they are able to hear the audio - but we are not able to see each other's videos.
Refreshing / Rejoining the meeting fixes this.
My preferred encoding in config.js H264 ( as using VP8 heats up the device a lot )
Since, Refresh / Rejoin solves the issue, but it is not user friendly - so can we make the code do this automatically?
Where do I need to make the code change?
I have done several customizations to the code, such as mute lock room etc - so will be able to do the above fix with some guidance. 

Here is my encoder config:
`		// Router media codecs.
			mediaCodecs :
			[
				{
					kind      : 'audio',
					mimeType  : 'audio/opus',
					clockRate : 48000,
					channels  : 2
				},
				{
					kind       : 'video',
					mimeType   : 'video/h264',
					clockRate  : 90000,
					parameters :
					{
						'packetization-mode'      : 1,
						'profile-level-id'        : '42E01e',
						'level-asymmetry-allowed' : 1,
						'x-google-start-bitrate'  : 1000
					}
				},

				{
					kind       : 'video',
					mimeType   : 'video/h264',
					clockRate  : 90000,
					parameters :
					{
						'packetization-mode'      : 1,
						'profile-level-id'        : '42e01f',
						'level-asymmetry-allowed' : 1,
						'x-google-start-bitrate'  : 1000
					}
				},

				{
					kind       : 'video',
					mimeType   : 'video/VP8',
					clockRate  : 90000,
					parameters :
					{
						'profile-id'             : 2,
						'x-google-start-bitrate' : 1000
					}
				},

				

				{
					kind       : 'video',
					mimeType   : 'video/h264',
					clockRate  : 90000,
					parameters :
					{
						'packetization-mode'      : 1,
						'profile-level-id'        : '42801e',
						'level-asymmetry-allowed' : 1,
						'x-google-start-bitrate'  : 1000
					}
				},

	
				{
					kind       : 'video',
					mimeType   : 'video/h264',
					clockRate  : 90000,
					parameters :
					{
						'packetization-mode'      : 1,
						'profile-level-id'        : '4d0032',
						'level-asymmetry-allowed' : 1,
						'x-google-start-bitrate'  : 1000
					}
				},

				{
					kind       : 'video',
					mimeType   : 'video/VP9',
					clockRate  : 90000,
					parameters :
					{
						'x-google-start-bitrate' : 1000
					}
				}

			]
	`

**Comment by jmencak** [comment]
> My preferred encoding in config.js H264 ( as using VP8 heats up the device a lot )
> Since, Refresh / Rejoin solves the issue, but it is not user friendly - so can we make the code do this automatically?
> Where do I need to make the code change?

I was wondering about the same thing recently, simply reordering the order of videocodecs in `config.js` did the trick for me if you want to be using H264.

**Comment by varundabke** [comment]
Yes, I have already re-ordered, as you can see in my original post. However, in some mobile devices, people have to refresh/rejoin to otherwise their videos are not visible to each-other in first attempt.
I am looking for a solution to this.

> > My preferred encoding in config.js H264 ( as using VP8 heats up the device a lot )
> > Since, Refresh / Rejoin solves the issue, but it is not user friendly - so can we make the code do this automatically?
> > Where do I need to make the code change?
> 
> I was wondering about the same thing recently, simply reordering the order of videocodecs in `config.js` did the trick for me if you want to be using H264.

**Comment by jmencak** [comment]
I can confirm I've hit this today too as a regular desktop user with Firefox 83 on Fedora 32.  The other party could not see my video until I rejoined.  @varundabke are you seeing this only on mobile devices?  Thanks.

**Comment by varundabke** [comment]
Was other party on Mobile?
Most of our users are on Mobile, so I am not sure if it occurs on desktop
as well - but dont remember noticing it on desktop.
Any expert comments from Edumeet team?

**Comment by jmencak** [comment]
> Was other party on Mobile? 

No, both were desktop Linux users.

**Comment by jmencak** [comment]
I wonder if this could be bandwidth related.  I've realized I was on a much slower ISP provider today and I haven't seen these issues in the past when using my regular/faster ISP.  I'll try to spend some time using traffic shaping to artificially limit my bandwidth and try and reproduce this.

**Comment by jmencak** [comment]
Linking an issue that could be related: https://github.com/edumeet/edumeet/issues/656

**Comment by varundabke** [comment]
for VP8, reducing frame rate has reduced battery consumption. After all the research, i have decided to go ahead with reduced frame rate + vp8

**Comment by jmencak** [comment]
I've just experienced the issue with video not being visible after joining on a desktop client (FF) audio worked.  Rejoining fixed the issue.  Will try to grab the logs next time if/when I see this. Running the latest `develop` branch.  I do not think this is related to #656 I linked above.
